language/move-lang/src/parser/ast.rs

```diff
@@ -134,6 +159,7 @@ pub struct FunctionSignature {
 pub enum FunctionVisibility {
     Public(Loc),
     Internal,
+    Spec(Loc), // spec only
```

Comment:
It feels VERY bizarre to me that spec functions are part of FunctionVisibility. A few reasons why:
- Being spec or not doesn't really feel like an issue of visibility. I get that it doesn't make sense for a function to be public and spec, but that case should be handled elsewhere. But grouping it here feels wrong since spec functions are really internal functions. In short, their visibility is internal, not "spec"
- The second reason this feels off is the spec functions will be removed at some point. But I currently use the same Visibility enum in every AST. I think having this information polluting the Visibility enum forever is wrong. So if we do keep it here (which I don't think we should), we should make a new Visibility enum that is used after they are gone

Reply:
Removed this from visibility